src/azul/queues.py

```diff
@@ -273,34 +241,22 @@ def limit_timeout(timeout):
             logger.info('Message count history (most recent first) is %r.',
                         list(reversed(total_lengths)))
 
-            if len(total_lengths) == total_lengths.maxlen and all(n == 0 for n in total_lengths) == empty:
-                logger.info('The queues are at the desired level.')
-                break
+            if len(total_lengths) == total_lengths.maxlen:
+                cummdiff = sum(
+                    abs(first - second)
+                    for first, second in zip(total_lengths, list(total_lengths)[1:])
+                )
+                if cummdiff * window_size * sleep_time / 60 <= epsilon:
```

Comment:
```suggestion
                if cummdiff <= epsilon:
```
Since `window_size` and `sleep_time` are hard coded in the function, couldn't this be simplified?

Reply:
Yes, but then this section would be invalid/buggy is either of those variables were changed. Do you think it would be better to rewrite this with the assumption those those values are going to stay fixed in the future?

Reply:
Hm. You are correct if `epsilon` is as defined in the docstring based on a per minute change. In that case I'd like to amend my suggestion by adding that `epsilon` should be based on the length of a full queue (`window_size * sleep_time`) instead of 1 minute, although I admit it would be a minor distinction in how `wait_to_stabilize` is used.

Reply:
I think I'll leave as-is and see what Hannes has to say.

Reply:
See my comment at the method signature.